### PR TITLE
Add release action

### PR DESCRIPTION
**Pull request by albertogdd**
## Description

Use pre-release workflow for final releases as well. `cosign.pub` is now attached to the release.

## How can this be tested?

Fork the repo and create a new release

Tested in my fork:
- [GH Action](https://github.com/albertogdd/alberto-dynatrace-operator/actions/runs/6196927449)
- [Release result](https://github.com/albertogdd/alberto-dynatrace-operator/releases/tag/v9.0.24)

## Checklist

~~- [ ] Unit tests have been updated/added~~
- [x] PR is labeled accordingly with a single label
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: hack/build/ci/generate-new-helm-index-yaml.sh [hunk 1]
```diff
@@ -17,6 +17,8 @@ trap 'cleanup' ERR
 
 cp "config/helm/repos/stable/index.yaml" "config/helm/repos/stable/index.yaml.previous"
 
+mkdir -p "${output_dir}"
```
Comment (andriisoldatenko) [inline hack/build/ci/generate-new-helm-index-yaml.sh:20]: why do we need this step?

Comment (andriisoldatenko) [reply hack/build/ci/generate-new-helm-index-yaml.sh:20]: but if the folder doesn't exist it means the previous step failed and the helm index won't work
am I missing something?

Comment (albertogdd) [reply hack/build/ci/generate-new-helm-index-yaml.sh:20]: The foldes is passed as parameter and so far in master branch is empty = always failing ?
https://github.com/Dynatrace/dynatrace-operator/blob/main/.github/workflows/release.yaml#L327

Comment (albertogdd) [reply hack/build/ci/generate-new-helm-index-yaml.sh:20]: I tried to fix it with the changes of this PR

Comment (andriisoldatenko) [reply hack/build/ci/generate-new-helm-index-yaml.sh:20]: I think it's another mess up with merge conflicts:
my version was https://github.com/andriisoldatenko/dynatrace-operator/blob/main/.github/workflows/release.yaml#L103

Comment (albertogdd) [reply hack/build/ci/generate-new-helm-index-yaml.sh:20]: Thanks! I will then use that as a reference :+1:

**Review by andriisoldatenko — APPROVED** [review]

**Pull request merged**